google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -102,15 +123,17 @@ def __init__(
         )
         self.context.load_verify_locations(cafile=self._ca_fileobject.name)
 
-
-class CloudSQLConnectionError(Exception):
-    """
-    Raised when the provided connection string is not formatted
-    correctly.
-    """
-
-    def __init__(self, *args: Any) -> None:
-        super(CloudSQLConnectionError, self).__init__(self, *args)
+    def get_preferred_ip(self, ip_types: List[IPTypes]) -> str:
+        """Returns the first IP address for the instance, in order of the
+        preference supplied by ip_types. If no IP addressess with the given
+        preference are found, an error is raised."""
+        for ip_type in ip_types:
+            if ip_type.value in self.ip_addrs:
+                return self.ip_addrs[ip_type.value]
+        raise CloudSQLConnectionError(
```

Comment:
The docstring for this error says:
```
Raised when the provided connection string is not formatted correctly.
```

We can probably address this in a different PR, but we probably need more descriptive error types.

Reply:
I can add a different error type for this. I also just opened up a new issue to add more error types and better error messages.